app/src/sharedTest/java/org/oppia/app/home/HomeActivityTest.kt

```diff
@@ -85,62 +94,90 @@ class HomeActivityTest {
     Intents.release()
   }
 
+  private fun setUpTestApplicationComponent() {
+    DaggerHomeActivityTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
   @Test
-  fun testHomeActivity_firstOpen_hasWelcomeString() {
+  fun testHomeActivity_recyclerViewIndex0_displayProfileName_profileNameDisplayedSuccessfully() {
     launch(HomeActivity::class.java).use {
       onView(
         atPositionOnView(
           R.id.home_recycler_view,
           0,
-          R.id.welcome_text_view
+          R.id.profile_name_textview
         )
-      ).check(matches(withText("Welcome to Oppia!")))
+      ).check(matches(withText("Sean!")))
     }
   }
 
   @Test
-  fun testHomeActivity_secondOpen_hasWelcomeBackString() {
-    simulateAppAlreadyOpened()
+  fun testHomeActivity_recyclerViewIndex0_displayGreetingMessageBasedOnTime_goodMorningMessageDisplayedSuccessful() {
+    launch(HomeActivity::class.java).use {
+      val c = Calendar.getInstance()
+      if (c.get(Calendar.HOUR_OF_DAY) in 5..11) {
+          onView(
+            atPositionOnView(
+              R.id.home_recycler_view,
+              0,
+              R.id.welcome_text_view
+            )
+          ).check(matches(withText("Good morning,")))
+        }
+    }
+  }
 
+  @Test
+  fun testHomeActivity_recyclerViewIndex0_displayGreetingMessageBasedOnTime_goodAfternoonMessageDisplayedSuccessful() {
     launch(HomeActivity::class.java).use {
-      // Wait until the expected text appears on the screen, and ensure it's for the welcome text view.
-      waitForTheView(withText("Welcome back to Oppia!"))
-      onView(
-        atPositionOnView(
-          R.id.home_recycler_view,
-          0,
-          R.id.welcome_text_view
-        )
-      ).check(matches(withText("Welcome back to Oppia!")))
+      val c = Calendar.getInstance()
+      if (c.get(Calendar.HOUR_OF_DAY) in 12..16) {
+          onView(
+            atPositionOnView(
+              R.id.home_recycler_view,
+              0,
+              R.id.welcome_text_view
+            )
+          ).check(matches(withText("Good afternoon,")))
+        }
```

Comment:
Add else condition in all test cases in which test case will fail everytime.

Reply:
I have changed this test. PTAL.